Library/Homebrew/dev-cmd/generate-formula-api.rb

```diff
@@ -55,6 +55,15 @@ def generate_formula_api
       tap_migrations_json = JSON.dump(tap.tap_migrations)
       File.write("api/formula_tap_migrations.json", tap_migrations_json) unless args.dry_run?
 
+      internal_formula_v3_hash = {
+        "tap_name"       => tap.name,
+        "tap_git_head"   => tap.git_head,
```

Comment:
```suggestion
        "tap"            => {
          "name"     => tap.name,
          "git_head" => tap.git_head,
        },
```

maybe?

Reply:
I went back and forth on that one a bit. Technically all of this information relates to a tap though so it seems kind of like unnecessary nesting.

Reply:
Maybe we should just name the file `formula_tap.json` and remove the tap prefix from all of the hash keys?

Reply:
Ok, I can see that argument that's fair

Reply:
> Maybe we should just name the file formula_tap.json

If anything it could be in reflection of the tap name so we have in theory homebrew-core.json, homebrew-cask.json and a (purely as an example) homebrew-cask-fonts.json would slot in nicely. Then there's probably no need to store the name in the JSON itself.

Reply:
> I don't think we even use the tap name when initializing formulas from the API so it might not be necessary to include anyway.

Probably not but if we move it to live in the tap class then it will probably be easier that way rather than saying "homebrew-core" = "formula" and "homebrew-cask" = "cask".

Reply:
Handled in https://github.com/Homebrew/brew/pull/16460/commits/bc029cfca31386bdb3764619956bac8d1fab31f0